lib/praxis/docs/open_api/schema_object.rb

```diff
@@ -39,12 +39,18 @@ def dump_schema(shallow: false, allow_ref: false)
               Praxis::Docs::OpenApiGenerator.instance.register_seen_component(type)
               h.merge!('$ref' => "#/components/schemas/#{type.id}")
             elsif @collection
-              items = OpenApi::SchemaObject.new(info: type.member_type).dump_schema(allow_ref: allow_ref, shallow: false)
+              items = OpenApi::SchemaObject.new(info: type.member_type).dump_schema(allow_ref: allow_ref, shallow: false, default_nullability: default_nullability)
               h = @attribute_options[:description] ? { 'description' => @attribute_options[:description] } : {}
               h.merge!(type: 'array', items: items)
             else # Attributor::Struct, etc
-              props = type.attributes.transform_values do |definition|
-                OpenApi::SchemaObject.new(info: definition).dump_schema(allow_ref: true, shallow: shallow)
+              props = type.attributes.transform_values.with_index do |definition, index|
+                # if type has an attribute in its requirements all, then it should be marked as required here
+                field_name = type.attributes.keys[index]
+                required_attributes = (type.describe[:requirements] || []).filter { |r| r[:type] == :all }.map { |r| r[:attributes] }.flatten.compact.uniq
```

Comment:
can you move this line above the transform loop? it's only type dependent, and can calculate it once.